newIDE/app/src/VariablesList/VariablesList.js

```diff
@@ -523,12 +524,23 @@ const VariablesList = (props: Props) => {
   );
 
   const [searchText, setSearchText] = React.useState<string>('');
-  const { onComputeAllVariableNames } = props;
+  const { onComputeAllVariableNames, onSelectedVariableChange } = props;
   const allVariablesNames = React.useMemo<?Array<string>>(
     () => (onComputeAllVariableNames ? onComputeAllVariableNames() : null),
     [onComputeAllVariableNames]
   );
-  const [selectedNodes, setSelectedNodes] = React.useState<Array<string>>([]);
+  const [selectedNodes, doSetSelectedNodes] = React.useState<Array<string>>([]);
+  const setSelectedNodes = React.useCallback(
+    (nodes: Array<string> | ((nodes: Array<string>) => Array<string>)) => {
+      nodes = Array.isArray(nodes) ? nodes : nodes(selectedNodes);
+      doSetSelectedNodes(nodes);
+      if (onSelectedVariableChange) {
+        onSelectedVariableChange(nodes);
+      }
+    },
+    [onSelectedVariableChange, selectedNodes]
```

Comment:
Do you really need to depend on `selectedNodes`? This means any state change will also change all the callbacks using `setSelectedNodes`. Previously, this was a stable function.

Reply:
Try this instead:

```
const [selectedNodes, doSetSelectedNodes] = React.useState<Array<string>>([]);
const setSelectedNodes = React.useCallback(
  (nodes: Array<string> | ((nodes: Array<string>) => Array<string>)) => {
    // Use the functional update form to access the current state directly
    doSetSelectedNodes(currentSelectedNodes => {
      const newNodes = Array.isArray(nodes) ? nodes : nodes(currentSelectedNodes);
      if (onSelectedVariableChange) {
        onSelectedVariableChange(newNodes);
      }
      return newNodes;
    });
  },
  [onSelectedVariableChange] // Remove selectedNodes from dependencies
);
```